code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -53,29 +66,58 @@
 
 	if(shock_stage >= 10) tally += 3
 
-
+	//No message required. See message below.
 	if(aiming && aiming.aiming_at) tally += 5 // Iron sights make you slower, it's a well-known fact.
 
-	if (drowsyness) tally += 6
+	if (drowsyness)
+		tally += 6
+		if(prob(messagechance))
+			messagechance = messagechance*0.5
+			src << "<span class='average'>Your fatigue slows you down.</span>"
 
 	if(FAT in src.mutations)
 		tally += 1.5
+		if(prob(messagechance*0.5)) // Being fat is more obvious so less the chance.
+			messagechance = messagechance*0.5
+			src << "<span class='average'>Your fat thighs chaff together, restricting your movement.</span>"
+
 	if (bodytemperature < 283.222)
 		tally += (283.222 - bodytemperature) / 10 * 1.75
+		if(prob(messagechance*0.5)) // Being hot is more obvious so less the chance.
+			messagechance = messagechance*0.5
+			src << "<span class='average'>You break out into a sweat and slow yourself down.</span>"
 
+	// Players don't need to be reminded of missing feat.
 	tally += max(2 * stance_damage, 0) //damaged/missing feet or legs is slow
+
+	//  Players don't need to be reminded of mutations as it is probably handled elsewhere.
 	if(mRun in mutations)
 		tally = 0
 
 	tally += move_delay_mod
 
+	//  Players don't need to be reminded of chem effects as messages are handled elsewhere.
 	if(tally > 0 && (CE_SPEEDBOOST in chem_effects))
 		tally = max(0, tally-3)
 
+	// Messaging should be handled elsewhere as tiles have different reasons as to why you're being slowed down.
 	var/turf/T = get_turf(src)
 	if(T)
 		tally += T.movement_cost
 
+	// Putting the hunger system here so being starving is MUCH more noticable as it now considers movement speed penalties
+	//Simpler hunger slowdown calculations, this should be a little faster due to no division, and more scaleable
+	if (nutrition < (max_nutrition * 0.4))
+		tally++
+		if(prob(messagechance))
+			messagechance = messagechance*0.5
+			src << "<span class='warning'>Your lack of nutrition makes movement difficult...</span>"
+	else if (nutrition < (max_nutrition * 0.1))
+		tally+= tally*2 //Yell at me if there is a *= function or something

```

Comment:
`//Yell at me if there is a *= function or something`

Pretty sure there is `x *= 2`, but that wouldn't be equivalent to what you wrote. `tally = tally + (tally * 2)` != `tally = tally * 2`